### ISSUE TITLE
fuzz test deterministic simulation

### ISSUE THREAD

**Issue by wangrunji0408** [risingwavelabs/risingwave#5010]
Our deterministic simulation e2e testing has been set up for a while. However, it didn't show a significant advantage in catching bugs compared to the normal test in a real cluster. Several problems found in parallel e2e testing did not appear in the simulation. The reason of this may be insufficient chaos injected. A basic fair random number generator is not powerful enough to quickly find those corner cases.
[One interesting idea from FoundationDB](https://youtu.be/0h9l6Ug_l5E?t=320) is to use fuzzing tool to generate random numbers in a simulation. Modern fuzzing tools utilize code coverage as a guide to generate high-quality inputs. Result from fdb shows that simulating with libfuzzer can find concurrent bugs much faster than without it. Looks like a promising solution we could try!

**Comment by wangrunji0408** [comment]
I just had a quick attempt [on madsim](https://github.com/madsim-rs/madsim/blob/2a555aeda734412ad578b4beb4756967314c41c4/madsim/fuzz/fuzz_targets/fuzz_target_1.rs), trying to reproduce the result from FoundationDB, but it seems not working at all.
I decide to put this on hold and may revisit later.

**Comment by fuyufjh** [comment]
Any updates?

**Comment by wangrunji0408** [comment]
Nope. We have no progress on it and won't be putting much effort into it recently. So just close this issue.
